lib/spack/spack/build_systems/autotools.py

```diff
@@ -415,6 +417,13 @@ def configure(self, spec, prefix):
         with working_dir(self.build_directory, create=True):
             inspect.getmodule(self).configure(*options)
 
+    def setup_build_environment(self, env):
+        if (self.spec.platform == 'darwin'
+                and macos_version() >= Version('11')):
+            # Many configure files rely on matching '10.*' for macOS version
+            # detection and fail to add flags if it shows as version 11.
+            env.set('MACOSX_DEPLOYMENT_TARGET', '10.16')
```

Comment:
Just curious, why `10.16` ? 
For NEURON we set `10.9`

Reply:
That's from upstream, I didn't really find documentation on this.

Reply:
Ok, looks like it's for config purposes only.